igor-monitor-plugins/src/main/java/com/netflix/spinnaker/igor/plugins/PluginsBuildMonitor.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2020 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.netflix.spinnaker.igor.plugins;
+
+import com.netflix.discovery.DiscoveryClient;
+import com.netflix.spectator.api.Registry;
+import com.netflix.spinnaker.igor.IgorConfigurationProperties;
+import com.netflix.spinnaker.igor.history.EchoService;
+import com.netflix.spinnaker.igor.plugins.front50.PluginReleaseService;
+import com.netflix.spinnaker.igor.plugins.model.PluginEvent;
+import com.netflix.spinnaker.igor.plugins.model.PluginRelease;
+import com.netflix.spinnaker.igor.polling.*;
+import com.netflix.spinnaker.security.AuthenticatedRequest;
+import java.time.Instant;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import lombok.Data;
+
+public class PluginsBuildMonitor
+    extends CommonPollingMonitor<
+        PluginsBuildMonitor.PluginDelta, PluginsBuildMonitor.PluginPollingDelta> {
+
+  private final PluginReleaseService pluginInfoService;
+  private final PluginCache cache;
+  private final Optional<EchoService> echoService;
+
+  public PluginsBuildMonitor(
+      IgorConfigurationProperties igorProperties,
+      Registry registry,
+      Optional<DiscoveryClient> discoveryClient,
+      Optional<LockService> lockService,
+      PluginReleaseService pluginInfoService,
+      PluginCache cache,
+      Optional<EchoService> echoService) {
```

Comment:
Curious - why is EchoService is optional?

Reply:
For deployments where you don't want to send notifications - e.g. passive regional clusters.